### PR TITLE
Add go-wrapper.sh script for disabling checklinkname in go1.23.0 and higher

### LINKED ISSUE #228: Add wrapper script for go test based on go version
I'm currently on go version `go1.23.1 linux/amd64` and I get the following error when trying to run `make test` or `make e2e`: `link: github.com/fjl/memsize: invalid reference to runtime.stopTheWorld`

However, running the tests directly with `go test` and the `-ldflags=-checklinkname=0` flag allows the tests to be run correctly.

We need to investigate creating a wrapper script for the go command that does the same check as in the `monogen.sh` script.

see https://discord.com/channels/1269403754588016816/1286002862626701353/1286030421280165991 for more details

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
Fixes #228

This pull request adds a new script, go-wrapper.sh, which is used to disable checklinkname in go1.23.0 and higher versions. This script is necessary because of a dependency on github.com/fjl/memsize. The script checks the Go version and sets the appropriate ldflags to disable checklinkname if the version is go1.23.0 or higher.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new shell script to manage Go command execution and ensure compatibility with Go version 1.23.0 and higher.
	- Standardized testing commands in the Makefile to utilize a new variable for enhanced consistency.
	- Added unit tests for the `ProofAPI` functionality within the Ethereum API, covering various scenarios and edge cases.

- **Bug Fixes**
	- Addressed compatibility issues with Go version 1.23.0 and higher through the wrapper script.

### CODE REVIEW

File: scripts/go-wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
```
Comment (natebeauregard) [inline scripts/go-wrapper.sh:2]: Could you add `set -e` in here as well to stop execution upon a command failure?

File: scripts/go-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+# Because we transitively depend on github.com/fjl/memsize, we need to disable checklinkname in go1.23.0 and higher.
+goVersion=$(go env GOVERSION)
+if [[ $goVersion > go1.23.0 || $goVersion == go1.23.0 ]]; then
+    exec go "$@" -ldflags=-checklinkname=0
+else
+    exec go "$@"
+fi
```
Comment (natebeauregard) [inline scripts/go-wrapper.sh:9]: nit: could you add a newline after the `fi`?

**Review by natebeauregard — COMMENTED** [review]
Looks good. Left two small nits but will approve/merge once they're addressed

**Review by natebeauregard — APPROVED** [review]
LGTM
